components/wmtk_components/marching/internal/Marching.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <wmtk/Scheduler.hpp>
+#include <wmtk/TriMesh.hpp>
+
+namespace wmtk::components::internal {
+
+class Marching
+{
+    TriMesh& m_mesh;
+    bool m_lock_boundary;
+
+    MeshAttributeHandle<double> m_position_handle; // record position
+    MeshAttributeHandle<long> m_vertex_tag; // record vertex tag value
+    MeshAttributeHandle<long> m_edge_tag; // record edge vertex tag value
+    long m_input_tag_value; // the value used to those simplicity you want to seperate
+    long m_embedding_tag_value; // the scalffold simplicities' tag value
+    long m_split_tag_value; // when you split a simplicity, you will set m_split_tag_value to the
+                            // new simplicity
+    int m_dimension; // operate dimension, 0 for vertex, 1 for edge, 2 for face, 3 for tet
+
+    Scheduler m_scheduler;
+
+    void process_in_2d();
+    void process_in_3d();
+
+public:
+    Marching(
+        TriMesh& mesh,
```

Comment:
This cannot be a `TriMesh` if this code should be able to handle also the 3D case.